plugins/jobtype/src/azkaban/jobtype/HadoopShell.java

```diff
@@ -0,0 +1,146 @@
+	public static final String HADOOP_GLOBAL_OPTS = "hadoop.global.opts";
+	public static final String WHITELIST_REGEX = "command.whitelist.regex";
+	public static final String BLACKLIST_REGEX = "command.blacklist.regex";
+
+	private HadoopSecurityManager hadoopSecurityManager;
+
+	public HadoopShell(String jobid, Props sysProps, Props jobProps, Logger log) throws RuntimeException {
+		super(jobid, sysProps, jobProps, log);
+
+		shouldProxy = getSysProps().getBoolean(HadoopSecurityManager.ENABLE_PROXYING, false);
+		getJobProps().put(HadoopSecurityManager.ENABLE_PROXYING, Boolean.toString(shouldProxy));
+		obtainTokens = getSysProps().getBoolean(HadoopSecurityManager.OBTAIN_BINARY_TOKEN, false);
+
+		if (shouldProxy) {
+			getLog().info("Initiating hadoop security manager.");
+			try {
+				hadoopSecurityManager = HadoopJobUtils.loadHadoopSecurityManager(getSysProps(), log);
+			} catch (RuntimeException e) {
+				e.printStackTrace();
+				throw new RuntimeException("Failed to get hadoop security manager!" + e.getCause());
+			}
+		}
+	}
+
+	@Override
+	public void run() throws Exception {
+		setupHadoopOpts(getJobProps());
+		HadoopConfigurationInjector.prepareResourcesToInject(getJobProps(), getWorkingDirectory());
+		if (shouldProxy && obtainTokens) {
+			userToProxy = getJobProps().getString("user.to.proxy");
+			getLog().info("Need to proxy. Getting tokens.");
+			Props props = new Props();
+			props.putAll(getJobProps());
+			props.putAll(getSysProps());
+
+			tokenFile = HadoopJobUtils.getHadoopTokens(hadoopSecurityManager, props, getLog());
+			getJobProps().put("env." + HADOOP_TOKEN_FILE_LOCATION, tokenFile.getAbsolutePath());
+		}
+		try {
+			super.run();
+		} catch (Exception e) {
+			e.printStackTrace();
+			throw new Exception(e);
+		} finally {
+			if (tokenFile != null) {
+				try {
+					HadoopJobUtils.cancelHadoopTokens(hadoopSecurityManager, userToProxy, tokenFile, getLog());
+				} catch (Throwable t) {
+					t.printStackTrace();
+					getLog().error("Failed to cancel tokens.");
+				}
+				if (tokenFile.exists()) {
+					tokenFile.delete();
+				}
+			}
+		}
+	}
+
+	/**
+	 * Append HADOOP_GLOBAL_OPTS with HADOOP_OPTS in the given props
+	 * 
+	 * @param props
+	 */
+	private void setupHadoopOpts(Props props) {
+		if (props.containsKey(HADOOP_GLOBAL_OPTS)) {
+			String hadoopGlobalOps = props.getString(HADOOP_GLOBAL_OPTS);
+			if (props.containsKey(HADOOP_OPTS)) {
+				String hadoopOps = props.getString(HADOOP_OPTS);
+				props.put(HADOOP_OPTS, String.format("%s %s", hadoopOps, hadoopGlobalOps));
+			} else {
+				props.put(HADOOP_OPTS, hadoopGlobalOps);
+			}
+		}
+	}
+
+	@Override
+	protected List<String> getCommandList() {
+		// Use the same parsing login as in default "command job";
+		List<String> commands = super.getCommandList();
+		return HadoopJobUtils.filterCommands(commands, getSysProps().getString(WHITELIST_REGEX, HadoopJobUtils.MATCH_ALL_REGEX), // ".*" will match everything
```

Comment:
Do you think it make sense to put the WHITELIST_REGEX and BLACKLIST_REGEX in this file instead of in HadoopJobUtils?
Putting them in HadoopJobUtils would only make sense if you expect maybe other job types might use them going forward